helix-tui/src/buffer.rs

```diff
@@ -510,9 +510,13 @@ impl Buffer {
     pub fn clear_with(&mut self, area: Rect, style: Style) {
         for x in area.left()..area.right() {
             for y in area.top()..area.bottom() {
-                let cell = &mut self[(x, y)];
-                cell.reset();
-                cell.set_style(style);
+                if let Some(cell) = &mut self.get_mut(x, y) {
+                    cell.reset();
+                    cell.set_style(style);
+                } else {
+                    // Skip this render if the window size has changed
+                    return;
+                }
```

Comment:
If I'm understanding it right, this index-out-of-bounds panic only happens because of the [underflow bug in `Rect::intersection`](https://github.com/helix-editor/helix/blob/a9973c6947382010b434564481dca9916294c84a/helix-view/src/graphics.rs#L240). Now that it's fixed, and `Rect::intersection` returns a zero-area rectangle instead of a huge rectangle outside of the viewport, this if/else check and early return will never even happen, since the loop has zero elements. Should we remove it?

Reply:
That's great. I'll close my PR

Reply:
Oops, I meant your PR is fixing the bug in `Rect::intersection`, and we should be merging instead of closing it :sweat_smile: Just pointing out that you don't have to change anything in `buffer.rs`.

Reply:
Not sure if `required_size` returning `None` is the cause of a separate crash, or if it's also because of `Rect::intersection` — I'd have to dig into the codebase more.

But seeing as the return type of `required_size` is an `Option<(u16, u16)>`, it's strange that returning `None` is an issue. Maybe the real bug is the calling function not handling `None` properly.